backend/core/views.py

```
@@ -5021,3 +5053,140 @@ def perform_destroy(self, instance):
+            task_identifier = (task_template_id, task_date)
+            
+            # Skip if we've already processed this task
+            if task_identifier in processed_tasks_identifiers:
+                continue
+                
+            # Check if this task should be processed (is in past or next 10)
+            if task in tasks_to_process:
+                processed_tasks_identifiers.add(task_identifier)
+                
+                # Get or create the TaskNode
+                task_node, created = TaskNode.objects.get_or_create(
```

Ruff (0.8.2)

`TaskNode` may be undefined, or defined from star imports

(F405)